src/erlcloud_ec2.erl

```diff
@@ -2936,6 +2946,221 @@ net_if_params(#ec2_net_if{}=X) ->
+                 ]};
+        {error, _} = Error ->
+            Error
+    end.
+
+%%
+%% Function for making calls to DeleteFlowLogs action
+%% DescribeFlowLogs Documentation: http://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DeleteFlowLogs.html
+%%
+-spec delete_flow_logs([string()]) -> ok_error(proplist()).
+delete_flow_logs(FlowIDs) ->
+    delete_flow_logs(FlowIDs, default_config()).
+
+-spec delete_flow_logs([string()], aws_config()) -> ok_error(proplist()).
+delete_flow_logs(FlowIDs, Config) when is_record(Config, aws_config) ->
+    {Resources, _} = lists:foldl(fun(ID, {Acc, Index}) ->
+                                    I = integer_to_list(Index),
+                                    FlowID = "FlowLogId."++I,
+                                    {[{FlowID, ID} | Acc], Index+1}
+                            end, {[], 1}, FlowIDs),
+
+    case ec2_query(Config, "DeleteFlowLogs", Resources, ?NEW_API_VERSION) of
+        {ok, Doc} ->
+            {ok, [
+                    {unsuccessful, [extract_unsuccesful_item(Item) || Item <- xmerl_xpath:string("/DeleteFlowLogsResponse/unsuccessful", Doc)]}
+                 ]
+            };
+        {error, _} = Error ->
+            Error
+    end.
+
+%%
+%% Function for making calls to DescribeFlowLogs action
+%% DescribeFlowLogs Documentation: http://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeFlowLogs.html
+%%
+-spec describe_flow_logs() -> ok_error(proplist()).
+describe_flow_logs() ->
+    describe_flow_logs([], [], default_config()).
+
+-spec describe_flow_logs(aws_config()) -> ok_error(proplist());
+                        (ec2_flow_ids()) -> ok_error(proplist());
+                        (filter_list()) -> ok_error(proplist());
+                        (ec2_max_result()) -> ok_error(proplist(), ec2_token()).
+describe_flow_logs(Config)
+    when is_record(Config, aws_config) ->
+    describe_flow_logs([], Config);
+describe_flow_logs([{_, _} | _] = Filter) ->
+    describe_flow_logs([], Filter, default_config());
+describe_flow_logs(FlowIDs)
+    when is_list(FlowIDs) ->
+    describe_flow_logs(FlowIDs, [], default_config());
+describe_flow_logs(MaxResults)
+    when is_integer(MaxResults) ->
+    describe_flow_logs(MaxResults, default_config()).
+
+-spec describe_flow_logs(ec2_flow_ids(), filter_list()) -> ok_error(proplist());
+                        (ec2_flow_ids(), aws_config()) -> ok_error(proplist());
+                        (filter_list(), aws_config()) -> ok_error(proplist());
+                        (filter_list(), ec2_max_result()) -> ok_error(proplist());
+                        (ec2_max_result(), aws_config()) -> ok_error(proplist(), ec2_token());
+                        (ec2_max_result(), ec2_token()) -> ok_error(proplist(), ec2_token()). 
+describe_flow_logs(FlowIDs, none = Filter) ->
+    describe_flow_logs(FlowIDs, Filter, default_config());
+describe_flow_logs(FlowIDs, [{_, _} | _] = Filter)
```

Comment:
this discussion was already held [here](https://github.com/alertlogic/erlcloud/pull/125#discussion-diff-81652895R187) and @motobob, @nalundgaard urged to consider using named types instead. Let's be consistent here as well.